aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -0,0 +1,861 @@
+    use aptos_std::bucket_table::{Self, BucketTable, new};
+    use std::bcs;
+
+    struct NFTMintConfig has key {
+        admin: address,
+        treasury: address,
+        signer_cap: SignerCapability,
+        token_minting_events: EventHandle<MintingEvent>,
+        token_exchange_events: EventHandle<ExchangeEvent>,
+    }
+
+    struct CollectionConfig has key {
+        collection_name: String,
+        collection_description: String,
+        collection_maximum: u64,
+        collection_uri: String,
+        collection_mutate_config: vector<bool>,
+        // this is base name, when minting, we will generate the actual token name as `token_name_base: sequence number`
+        token_name_base: String,
+        royalty_payee_address: address,
+        token_description: String,
+        token_maximum: u64,
+        token_uri: vector<String>,
+        token_mutate_config: TokenMutabilityConfig,
+        royalty_points_den: u64,
+        royalty_points_num: u64,
+        property_keys: vector<vector<String>>,
+        property_values: vector<vector<vector<u8>>>,
+        property_types: vector<vector<String>>,
+    }
+
+    struct WhitelistMintConfig has key {
+        whitelisted_address: BucketTable<address, u64>,
+        whitelist_mint_price: u64,
+        whitelist_minting_start_time: u64,
+        whitelist_minting_end_time: u64,
+    }
+
+    struct PublicMintConfig has key {
+        public_mint_price: u64,
+        public_minting_start_time: u64,
+        public_minting_end_time: u64,
+    }
+
+    struct SourceToken has key {
+        token_data_id: TokenDataId,
+    }
+
+    struct MintingEvent has drop, store {
+        token_receiver_address: address,
+        token_id: TokenId,
+    }
+
+    struct ExchangeEvent has drop, store {
+        token_receiver_address: address,
+        token_id: TokenId,
+    }
+
+    const BURNABLE_BY_OWNER: vector<u8> = b"TOKEN_BURNABLE_BY_OWNER";
+
+    const ENOT_AUTHORIZED: u64 = 1;
+    const EINVALID_TIME: u64 = 2;
+    const EACCOUNT_DOES_NOT_EXIST: u64 = 3;
+    const EVECTOR_LENGTH_UNMATCHED: u64 = 4;
+    const EEXCEEDS_COLLECTION_MAXIMUM: u64 = 5;
+    const EINVALID_PRICE: u64 = 6;
+    const EINVALID_UPDATE_AFTER_MINTING: u64 = 7;
+    const EMINTING_IS_NOT_ENABLED: u64 = 8;
+    const ENO_ENOUGH_TOKENS_LEFT: u64 = 9;
+    const EACCOUNT_NOT_WHITELISTED: u64 = 10;
+    const EINVALID_ROYALTY_NUMERATOR_DENOMINATOR: u64 = 11;
+    const ECOLLECTION_ALREADY_CREATED: u64 = 12;
+    const ECONFIG_NOT_INITIALIZED: u64 = 13;
+    const EAMOUNT_EXCEEDS_MINTS_ALLOWED: u64 = 14;
+    const ETOKEN_ID_NOT_FOUND: u64 = 15;
+    const ECANNOT_EXCHANGE_BEFORE_MINTING_ENDS: u64 = 16;
+
+    fun init_module(resource_account: &signer) {
+        let resource_signer_cap = resource_account::retrieve_resource_account_cap(resource_account, @source_addr);
+        move_to(resource_account, NFTMintConfig {
```

Comment:
What's the relation between resource_account and post_mint_reveal_nft? Other functions fetch NFTMintConfig from post_mint_reveal_nft but this is creating it under the resource account? What happens if they're not the same?